examples/java/hello_world/MainActivity.java

```diff
@@ -53,7 +54,9 @@ protected void onCreate(Bundle savedInstanceState) {
       Log.d("MainActivity", "exception getting config.", e);
       throw new RuntimeException("Can't get config to run envoy.");
     }
-    envoy = new Envoy(context, config);
+
+    EnvoyEngine envoyEngine = new AndroidEngineImpl(context);
```

Comment:
At the moment, this will have to do. A more ideal ergonomic solution is to use a Builder or Factory to create an `Envoy` instance

Reply:
Spoke with @rebello95 offline. We're going to go with the Builder pattern.

His [PR](https://github.com/lyft/envoy-mobile/pull/328) introduces a `Configuration` concept for Envoy. We'll rename the class and have that responsible for creating Envoy instances with the user specified configuration

Reply:
@buildbreaker will update this to look like iOS: https://github.com/lyft/envoy-mobile/pull/343